### PR TITLE
Style required field error messages consistently

### PR DESCRIPTION
**Pull request by RoyEJohnson**
Per
http://2jsgoo.axshare.com/#g=1&p=assignment_builder_style_bug_fixes&c=1

### CODE REVIEW

**Comment by RoyEJohnson** [comment]
[Bug: Error state for required fields in tutor assignment builder should not have solid orange background, but orange line under text field instead](https://trello.com/c/Q6PlqqmI/310-bug-error-state-for-required-fields-in-tutor-assignment-builder-should-not-have-solid-orange-background-but-orange-line-under-te)

**Comment by RoyEJohnson** [comment]
[Make styling, capitalization/punctuation, and wording for required field errors consistent in Tutor assignment builders.](https://trello.com/c/WRTGN7bH/290-make-styling-capitalization-punctuation-and-wording-for-required-field-errors-consistent-in-tutor-assignment-builders)

File: tutor/src/components/task-plan/footer/index.cjsx [hunk 1]
```diff
@@ -102,9 +102,6 @@ PlanFooter = React.createClass
         isEditable={@state.isEditable}
         getBackToCalendarParams={@props.getBackToCalendarParams}
       />
-      <HelpTooltip
-        isPublished={isPublished}
-      />
```
Comment (pandafulmanda) [inline tutor/src/components/task-plan/footer/index.cjsx]: sorry i just got word from ux that they want to keep this, at least for now

File: tutor/src/components/task-plan/reading.cjsx [hunk 2]
```diff
@@ -161,15 +161,14 @@ ReadingPlan = React.createClass
 
     if not @state.isVisibleToStudents
       addReadingsButton = <BS.Button id='reading-select'
-        className="-select-sections-btn"
+        className={"-select-sections-btn" + (if @state?.invalid and not topics?.length then ' invalid')}
```
Comment (pandafulmanda) [inline tutor/src/components/task-plan/reading.cjsx]: `classnames` could be used here as well

File: tutor/src/components/tutor-input.cjsx [hunk 3]
```diff
@@ -167,10 +167,6 @@ TutorDateInput = React.createClass
     @props.onChange(value)
     @setState({expandCalendar: false, valid, value, errors})
 
-  # TODO There's a bug in our version of datepicker where onBlur fires when
-  #   changing months.  Put this onBlur back as a prop for Datepicker when
-  #   we upgrade so that label and invalid date errors can display properly
-  #   on blur.
```
Comment (pandafulmanda) [inline tutor/src/components/tutor-input.cjsx]: nice catch, thanks for updating this and the `onBlur`!

File: tutor/resources/styles/components/task-plan/builder.less [hunk 4]
```diff
@@ -75,14 +75,24 @@
...
   margin: -10px 0 30px;
 }
 
 .due-before-open,
 .readings-required,
-.problems-required {
+.problems-required,
+.required-hint {
   color: @form-error-color;
   font-size: 1.2rem;
```
Comment (pandafulmanda) [inline tutor/resources/styles/components/task-plan/builder.less]: for `.problems-required` and `.readings-required`, there's a discrepancy between current and mockup:

## now:

screen shot 2016-12-09 at 10 06 52 am

## mockup:

screen shot 2016-12-09 at 10 07 00 am

**Comment by nathanstitt** [comment]
I'm sorry @RoyEJohnson, @pandafulmanda had asked me to hold off merging #1481 because it would conflict with your work.  I then promptly forgot about her request while I tested it and merged it anyway .

**Comment by RoyEJohnson** [comment]
No problem, @nathanstitt. I rebased to master and only had one conflict. I blended in your change (homework.cjsx lines 68-70)

**Comment by pandafulmanda** [comment]
looks great!

**Pull request merged**